bpf/lib/bpf_cgroup.h

```diff
@@ -10,16 +10,91 @@
 
 #define NULL ((void *)0)
 
+#ifndef CGROUP_SUPER_MAGIC
+#define CGROUP_SUPER_MAGIC 0x27e0eb /* Cgroupv1 pseudo FS */
+#endif
+
+#ifndef CGROUP2_SUPER_MAGIC
+#define CGROUP2_SUPER_MAGIC 0x63677270 /* Cgroupv2 pseudo FS */
+#endif
+
 /* Our kernfs node name length, can be made 256? */
 #define KN_NAME_LENGTH 128
 
+/* Max nested cgroups that are tracked. Arbitrary value, nested cgroups
+ * that are at a level greater than 32 will be attached to the cgroup
+ * at level 32.
+ */
+#define CGROUP_MAX_NESTED_LEVEL 32
+
+typedef enum {
+	CGROUP_UNTRACKED = 0, /* Cgroup was created but we did not track it */
+	CGROUP_NEW = 1, /* Cgroup was just created */
+	CGROUP_RUNNING = 2, /* new => running (fork,exec task inside) */
+	CGROUP_RUNNING_PROC = 3, /* Generated from pids of procfs */
+} cgroup_state;
+
 /* Represent old kernfs node with the kernfs_node_id
  * union to read the id in 5.4 kernels and older
  */
 struct kernfs_node___old {
 	union kernfs_node_id id;
 };
 
+struct cgroup_tracking_value {
+	/* State of cgroup */
+	cgroup_state state;
```

Comment:
Hmm yes my memory tricked me... enums can be small, however for our case and targeted platform as discussed it will always be 4 bytes and we don't do short enums anyway, also we have that align checker, so I think we are good.